src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -201,6 +202,21 @@ suggestRemoveRedundantImport ParsedModule{pm_parsed_source = L _  HsModule{hsmod
         = [("Remove import", [TextEdit (extendToWholeLineIfPossible contents _range) ""])]
     | otherwise = []
 
+removeAllRedundantImports :: Maybe ParsedModule -> Maybe T.Text -> [Diagnostic] -> Uri -> [CAResult]
+removeAllRedundantImports m contents diagnostics uri
+  | Just pm <- m,
+    r <- join $ map (suggestRemoveRedundantImport pm contents) diagnostics,
+    tes <- nub [ e | (_, edits) <- r, e <- edits],
+    _changes <- Just $ Map.singleton uri $ List tes,
+    _title <- "Remove all redundant imports",
+    _kind <- Just CodeActionQuickFix,
+    _diagnostics <- Nothing,
+    _documentChanges <- Nothing,
+    _edit <- Just WorkspaceEdit{..},
+    _command <- Nothing
+   =  [CACodeAction CodeAction {..}]
```

Comment:
Will the code action show when there are no redundant imports?

Reply:
Added a `not null` guard.